cmd/build.go

```diff
@@ -154,7 +163,56 @@ func picardImage() (string, error) {
 	return fmt.Sprintf("%s@%s", picardRepo, sha), nil
 }
 
+func configVersion(configBytes []byte) (string, bool) {
+	var raw map[string]interface{}
+	if err := yaml.Unmarshal(configBytes, &raw); err != nil {
+		return "", false
+	}
+
+	var configWithVersion struct {
+		Version string `yaml:"version"`
+	}
+	if err := mapstructure.WeakDecode(raw, &configWithVersion); err != nil {
+		return "", false
+	}
+	return configWithVersion.Version, true
+}
+
+func validateConfigVersion(args []string) error {
+	invalidConfigError := "\nYou attempted to run a local build with version '%s' of configuration.\nLocal builds do not support that version at this time.\nYou can use 'circleci config process' to pre-process your config into a version that local builds can run (see 'circleci help config process' for more information)"
```

Comment:
To make this string a bit more readable you could use golang string literals (https://golang.org/ref/spec#String_literals) to format it accross multiple strings. It's hard to show an example of this in github because github markdown uses the same character for inline code blocks, but here's a gist:  https://gist.github.com/dnephin/694496562771a68e34be9cab559ee598